### PR TITLE
feat: Handle permission driven views for auto-saving pages and action

### LINKED ISSUE #16458: [Task]: Permission-driven views for Actions

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

> Updated views for pages/actions if the user doesn't have edit permission for the same. This will lead to showing a 403 forbidden error toast message, mentioning that the changes are not being saved.

Fixes #16458 #16459

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Tested it locally.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/editorComponents/EditableText.tsx [hunk 1]
```diff
@@ -217,14 +220,18 @@ export function EditableText(props: EditableTextProps) {
...
-              fillColor="#939090"
-              name="edit"
-              size={IconSize.XXL}
-            />
-          )}
+          {!disabled &&
+            !minimal &&
+            !hideEditIcon &&
+            !updating &&
+            !isEditing && (
```
Comment (berzerkeer) [inline app/client/src/components/editorComponents/EditableText.tsx:227]: Can we extend this to a variable to make it more readable here ?

File: app/client/src/selectors/workspaceSelectors.tsx [hunk 2]
```diff
@@ -1,68 +0,0 @@
-import { createSelector } from "reselect";
```
Comment (berzerkeer) [inline app/client/src/selectors/workspaceSelectors.tsx:1]: Can you let me know why we have delete this file ?

**Review by berzerkeer — COMMENTED** [review]
Requested few changes.

File: app/client/src/selectors/workspaceSelectors.tsx (hunk 2 above)
Comment (ankitakinger) [reply app/client/src/selectors/workspaceSelectors.tsx:1]: This file was split. This is older one which was mistakenly left in the codebase.

File: app/client/src/components/editorComponents/EditableText.tsx (hunk 1 above)
Comment (ankitakinger) [reply app/client/src/components/editorComponents/EditableText.tsx]: Done

**Review by berzerkeer — APPROVED** [review]

**Pull request merged**